packages/quantification/src/net-quantification.ts

```diff
@@ -0,0 +1,185 @@
+
+  let rowIndex = 0;
+  let colIndex = 0;
+  let debt = 0;
+  while (rowIndex < netQuantifications.length) {
+    while (colIndex < yearsOrderedAsc.length) {
+      const yearIndex = yearsOrderedAsc[colIndex];
+      const value = netQuantifications[rowIndex][yearIndex];
+      logger?.debug(`Visiting [${rowIndex}, ${yearIndex}] = ${value}`);
+
+      if (value < 0) {
+        logger?.debug(`Found negative number (${value}), adding to debt`);
+        debt += value;
+        netQuantifications[rowIndex][yearIndex] = 0;
+      }
+
+      if (debt < 0) {
+        logger?.debug(`Accounting for debt of ${debt}`);
+        // Edge case: if there's only one year, start on the next field, otherwise start on the current
+        let accountingRowIndex =
+          yearsOrderedAsc.length === 1
+            ? rowIndex + (1 % netQuantifications.length)
+            : rowIndex;
+        let accountingColIndex = (colIndex + 1) % years.size;
+        let accountingYearIndex = yearsOrderedAsc[accountingColIndex];
+        while (debt < 0) {
+          const accountingCellValue =
+            netQuantifications[accountingRowIndex][accountingYearIndex];
+
+          logger?.debug(
+            `Accounting visiting [${accountingRowIndex}, ${accountingYearIndex}] = ${accountingCellValue}`
+          );
+
+          // Core logic: the amount to take is the lesser of the absolute value of the debt or the
+          // the amount available
+          const amountAvailable = Math.max(accountingCellValue, 0);
+          const amountToTake = Math.min(amountAvailable, Math.abs(debt));
+          const cellRemainder = accountingCellValue - amountToTake;
+          netQuantifications[accountingRowIndex][accountingYearIndex] =
+            cellRemainder;
+          debt += amountToTake;
+
+          logger?.debug(
+            `Removing ${amountToTake} from [${accountingRowIndex}, ${accountingYearIndex}] = ${cellRemainder}`
+          );
+          logger?.debug(`Debt left: ${debt}`);
+          logger?.table(netQuantifications);
+
+          // Go to the next column
+          accountingColIndex =
+            (accountingColIndex + 1) % yearsOrderedAsc.length;
+          accountingYearIndex = yearsOrderedAsc[accountingColIndex];
+          logger?.debug(`Advancing to next year (col ${accountingYearIndex})`);
+
+          // If we're back to the row's starting year, advance to the next field
+          if (accountingColIndex === colIndex) {
+            accountingRowIndex =
+              (accountingRowIndex + 1) % netQuantifications.length;
+            logger?.debug(
+              `Advancing to next field (row ${accountingRowIndex})`
+            );
+          }
+
+          // If we get back to where we started, we can't account for the debt
+          // Set the cell's value to the remainder, clear the debt and break
+          if (
+            accountingRowIndex === rowIndex &&
+            accountingColIndex === colIndex
+          ) {
+            logger?.debug(
```

Comment:
Love these optional debugger statements and generally how you're illustrating with documentation along the way. Definitely helps to reduce the brain strain, I'm sure as much for development as it is for review and will be for maintenance.

Reply:
Yeah, it was hugely helpful during development. I figure it's an important enough algorithm that we might want to check the output every now and then. Passing in a logger, was can filter out the debug statements if it's too noisy.

Reply:
+1; would love to consolidate our approach to logging at some point